OPSC-17715 OpsCenter 6.8.45 Release (#213)
datastax/release-notes: OpsCenter_6.8_Release_Notes.md

# Release notes for OpsCenter

# Release Notes for OpsCenter 6.8.45
17th June,2025

## Core

> * Added the {{node_ips}} option to monitor nodes for events. Setting this field limits alerts monitoring to the nodes specified in the list. For example: ["127.0.0.1","127.0.0.2"]. (OPSC-17644)

Reviewer: I'm still looking for it to be clarified that it is "alert" rather than "event" configuration.

> * *Targeted Solr Core Fetching:* OpsCenter now optimizes how Solr core information is retrieved. Instead of fetching cores for _all_ keyspaces and tables, OpsCenter will now fetch only the cores associated with the {{selected keyspaces and tables}}. This improves performance and reduces unnecessary data processing. (OPSC-17704)

Reviewer 2: ```suggestion
* Improved the Solr core fetching process to be more targeted. OpsCenter now optimizes the retrieval of Solr core information. Instead of fetching cores for all keyspaces and tables, OpsCenter now fetches only the cores associated with selected keyspaces and tables. This improves performance and reduces unnecessary data processing. (OPSC-17704)
```

Reviewer 2: According to our documentation style, every Release Notes entry needs to start with a past-tense verb (Fixed, Added, Updated, etc.). The rest of the sentences need to be in present tense to indicate how the new functionality now works. Also, not sure you really need to include all the extra formatting (bold and italic) unless you need to specify emphasis.

* Added enhancements to authentication configuration and user lockout management. (OPSC-15418)
* Upgraded the {{org.json:json}} library to version {{20230227}}. (OPSC-17628 OPSC-17629)
* Added new options to the {{authentication}} section of {{opscenterd.conf}} to allow users to use {{password_db}} and {{passwd_db}} alias. (OPSC-17701)

> * Improved data insertion logic to handle duplicates, enhancing reliability and preventing insertion errors. (OPSC-17693)

Reviewer: What's the feature? A customer would have no idea what this is referring to. Add  "user role"  somewhere in there.

## Backup Service

> * OpsCenter now compresses only the *differential SSTables* *when syncing a snapshot to a destination* and *compression is enabled*. Compression is applied *after* the differential SSTable calculation, significantly reducing temporary storage usage preventing temporary storage runout issues (OPSC-17694)

Reviewer 2: ```suggestion
* Revised the compression process. OpsCenter now compresses only the differential SSTables when you sync a snapshot to a destination and enable compression. The system applies compression after the differential SSTable calculation, significantly reducing temporary storage usage and preventing temporary storage runout issues. (OPSC-17694)
```

Reviewer 2: Another documentation style tip: Try to avoid using passive language. The goal is to identify the person or process that performs and action, and then assign the action/verb to this person or process. Easiest way to spot passive language? Look for the words "is" and "are." Eliminate these and you're in good shape.

# Release notes for OpsCenter

# Release Notes for OpsCenter 6.8.45
17th June,2025

## Core
* {{node_ips}} Nodes to monitor for events. Setting this field limits event monitoring to the nodes specified in the list. Example: ["127.0.0.1","127.0.0.2"] (OPSC-17644)
* inclusion of boolean `ignore_solr_resources` variable under `backup_service` section in `opscenterd.conf` (OPSC-17704)

> * Enhancements to Authentication Configuration and User Lockout Management (OPSC-15418)

Reviewer 2: ```suggestion
* Introduced the following enhancements to authentication configuration and user lockout management:
  * Added the `permanent_lockout_limit` option to the `authentication` section of the `opscenterd.conf` configuration file. This option supports permanent user lockouts. You can configure the number of total failed login attempts before the system locks out the user permanently. By default, this option is disabled.
  * Introduced two new API endpoints to manage permanently locked out users:
    * `GET /users/permanent-lockouts`: This option retrieves a list of users who are permanently locked out.
    * `GET /users/**USERNAME**/reset-permanent-lockout`: This option resets the permanent lockout status of a specified user. Only administrators can perform this operation.
```

* Upgraded the {{org.json:json}} library to version {{20230227}}. (OPSC-17628 OPSC-17629)
* Added new options to the `authentication` section of `opscenterd.conf` to allow users to use `password_db` and `passwd_db` alias. (OPSC-17701)
* Improved data insertion logic to handle duplicates, enhancing reliability and preventing insertion errors. (OPSC-17693)

## Backup Service

> * The compression logic is shifted to the point after the `differential ss-tables` are calculated, optimizing the temporary storage runout issues. (OPSC-17694)

Reviewer 2: ```suggestion
* Changed OpsCenter to compress only the differential SSTables when you enable compression. The system now performs compression after the differential SSTable calculation occurs. This enhancement optimizes temporary storage runout issues. (OPSC-17694)
```

Reviewer: Again the comment here lacks a clear indication of what feature it is associated with. We need to add that this is when syncing a snapshot to a destination. This is coming up again and again so I'll post a note to the opscenter-eng room too.

## UI

> * Customers can now selectively generate `diagnostics` reports for specific datacenters or nodes directly from the OpsCenter UI, reducing report size and improving performance when working with large clusters. (OPSC-17610)

Reviewer 2: ```suggestion
* Provided the ability within the OpsCenter UI to generate diagnostics reports selectively for specific datacenters or nodes. This feature reduces the report size and improves performance when working with large clusters. (OPSC-17610)
```
